beginner.md

```diff
@@ -0,0 +1,588 @@
+# delite - creating custom components
+
+## delite background
+`delite` is a new JavaScript library born out of the [Dojo Toolkit Dijit framework](http://dojotoolkit.org/reference-guide/1.10/dijit).
+This isn't a replacement per se but a repository to be used as the core building blocks for leveraging current and future standards
+in HTML, CSS & JavaScript to build reusable Web Components.
+
+It can be used on its own but more likely used with other projects either from the [ibm-js repositories](https://github.com/ibm-js)
+or other repositories.
+
+More information can be found on the [delite website](http://ibm-js.github.io/delite/) explaining the standards this library aims to conform to.
+
+## Tutorial details
+In this tutorial you'll learn how to create your own custom elements, learn how to register them, learn how to use templates
+and learn how to theme them. It's a beginner tutorial so we won't be delving too deep into what `delite` provides (yet!!!).
+
+
+## Getting started
+To quickly get started, we're using [https://github.com/ibm-js/generator-delite-element](https://github.com/ibm-js/generator-delite-element)
+to install the required dependencies and create a basic scaffold.
+
+Install the `generator-delite-element` globally (or update it if necessary).
+
+    npm install -g generator-delite-element
+
+##Templates
+`delite` provides first class support for templates. We wouldn't expect to programmatically create DOM nodes & this is where
+`delite` comes into it's own; out of the box `delite` supports templating using a built in implementation of [Handlebars](http://handlebarsjs.com/).
+
+Note there are some limitations using the `delite/handlebars!` plugin for templating, namely it doesn't support iterators or conditionals.
+However in many cases this isn't a limiting factor. Support for this will be explained in a later more advanced tutorial when we discuss
+[Liaison](https://github.com/ibm-js/liaison). The handlebars template implementation `delite` uses is primarily focused on performance.
```

Comment:
maybe mention quickly that alternate templating engine can be plugged if needed (that's why liaison will be working). I guess people can understanding it reading between the lines but better say it explicitly.